dali/dfu/dfurun.cpp

```diff
@@ -1246,20 +1250,18 @@ class CDFUengine: public CInterface, implements IDFUengine
                             // need to set ftslave location
                             StringBuffer progpath;
                             StringBuffer workdir;
-                            INode *n = srcFile->getFileDescriptor()->queryNode(0);
+                            INode *n = srcFdesc->queryNode(0);
                             if (n&&getRemoteRunInfo("FTSlaveProcess", "ftslave", NULL, n->endpoint(), progpath, workdir, foreigndalinode, 1000*60*5))
                             {
                                 opttree->setProp("@slave",progpath.str());
                             }
                         }
-
                     }
                     if (destination->getMultiCopy()&&!destination->getWrap())
                     {
                         Owned<IFileDescriptor> tmpfd = srcFile->getFileDescriptor();
```

Comment:
since you have srcFDesc in scope now, this can be replaced too.

Reply:
Changed